### DISCUSSION TITLE
Potential use cases?

### DISCUSSION THREAD

**Discussion by pascalandy** [stefanzweifel/git-auto-commit-action#4] [Help] [answered]
Hi, I'm reading everything I can about Github actions. 

Now, I'm curious to understand the use case of this action. 

Thanks!

**Comment by owenvoke** [comment]
I guess maybe if you updated source files, wanted to compile/minify/etc. them and add the results to the repository. For example, with a `src` / `dist` directory structure.

**Comment by stefanzweifel** [comment, marked as answer]
Building and pushing your CSS/JS artifacts back to the repo is one use case.

Another use case is running a linter or code-style-checker (like Prettier or php-cs-fixer) on GitHub Actions and then commit the changed/fixed files back to the repository. It's basically the same as adding https://styleci.io or similar services to your projects.

I've added an example workflow on how to use php-cs-fixer and this Action in the README:
https://github.com/stefanzweifel/git-auto-commit-action#example-usage

Yet another use case for me, is creating a new build for a static site. 
In this project, I want GitHub Actions to run my build script at the first day of the month. Action will clone the project, run a script to update a sqlite-database, run a script to create a HTML-build and commit all changes back to the repository.
Netlify will then automatically deploy my app.
https://github.com/stefanzweifel/laravel-download-statistics-app/blob/master/.github/workflows/schedule.yml

---

I know that many people despise the idea of committing build files to the repository. I personally think, that this is fine for smaller projects where a big deploy pipeline would be an overkill.

**Comment by owenvoke** [comment]
That's a really neat idea actually (about running the linter and re-committing). :thinking: I use Style CI for open source, but for some of my private projects it would be good to have an equivalent. Thanks for the idea. :+1:

**Comment by stefanzweifel** [comment]
What I couldn't figure out yet, if this idea of running the linter on GitHub Actions also works for forks or pull requests from contributors.

The best constellation for me would be, to run the linter for my commits and for the commits of a proposed PR. That way the person creating the PR wouldn't have to think about code style issues as the Action would take care of that.

Something I wanna test over the next few days …

**Comment by pascalandy** [comment]
@stefanzweifel Thanks for you detailed answer. I like both use cases.

I actually have heavy-duty bash scripts that auto-commit many logs from my clusters :-p
